source/CompactRegular.lagda

```diff
@@ -0,0 +1,402 @@
+compacts-are-closed-under-joins F U V κ₁ κ₂ S dir@(_ , up) p =
+ ∥∥-rec₂ ∃-is-prop γ s₁′ s₂′
+  where
+   open PosetNotation  (poset-of F) using (_≤_)
+   open PosetReasoning (poset-of F)
+
+   -- S covers U.
+   q₁ : (U ≤ (⋁[ F ] S)) holds
+   q₁ = U ≤⟨ ∨[ F ]-upper₁ U V ⟩ U ∨[ F ] V ≤⟨ p ⟩ ⋁[ F ] S ■
+
+   -- S covers V.
+   q₂ : (V ≤ (⋁[ F ] S)) holds
+   q₂ = V ≤⟨ ∨[ F ]-upper₂ U V ⟩ U ∨[ F ] V ≤⟨ p ⟩ ⋁[ F ] S ■
+
+   s₁′ : ∥ Σ i₁ ꞉ index S , (U ≤ S [ i₁ ]) holds ∥
+   s₁′ = κ₁ S dir q₁
+
+   s₂′ : ∥ Σ i₂ ꞉ index S , (V ≤ S [ i₂ ]) holds ∥
+   s₂′ = κ₂ S dir q₂
+
+   γ : (Σ i₁ ꞉ index S , (U ≤ S [ i₁ ]) holds)
+     → (Σ i₂ ꞉ index S , (V ≤ S [ i₂ ]) holds)
+     → ∃ i ꞉ index S , ((U ∨[ F ] V) ≤ S [ i ]) holds
+   γ (i₁ , s₁) (i₂ , s₂) = ∥∥-rec ∃-is-prop δ (up i₁ i₂)
+    where
+     δ : Σ i ꞉ index S , ((S [ i₁ ] ≤ S [ i ]) ∧ (S [ i₂ ] ≤ S [ i ])) holds
+       → ∃ i ꞉ index S , ((U ∨[ F ] V) ≤ S [ i ]) holds
+     δ (i , r₁ , r₂) = ∣ i , ∨[ F ]-least ε ζ ∣
+      where
+       ε : (U ≤ S [ i ]) holds
+       ε = U ≤⟨ s₁ ⟩ S [ i₁ ] ≤⟨ r₁ ⟩ S [ i ] ■
+
+       ζ : (V ≤ S [ i ]) holds
+       ζ = V ≤⟨ s₂ ⟩ S [ i₂ ] ≤⟨ r₂ ⟩ S [ i ] ■
+
+\end{code}
+
+\section{Well Inside}
+
+\begin{code}
+
+-- The type family expressing that a given open of a frame is clopen.
+well-inside₀ : (F : frame 𝓤 𝓥 𝓦) → ⟨ F ⟩ → ⟨ F ⟩ → 𝓤 ̇ 
+well-inside₀ F U V =
+ Σ W ꞉ ⟨ F ⟩ , (U ∧[ F ] W ≡ 𝟎[ F ]) × (V ∨[ F ] W ≡ 𝟏[ F ])
+
+infix 4 well-inside₀
+
+syntax well-inside₀ F U V = U ⋜₀[ F ] V
+
+-- The well inside relation is not propositional in general, even though the
+-- “has complement” (i.e. is well inside itself) is propositional.
+well-inside-is-not-prop : is-univalent 𝓤₀
+                        → Σ F ꞉ frame 𝓤₁ 𝓤₀ 𝓤₀ ,
+                           (¬ ((U V : ⟨ F ⟩) → is-prop (U ⋜₀[ F ] V)))
+well-inside-is-not-prop ua = IF , ε
+ where
+  IF : frame 𝓤₁ 𝓤₀ 𝓤₀ -- “IF” standing for “initial frame”.
+  IF = 𝟎-𝔽𝕣𝕞 ua
+
+  γ₂ : 𝟎[ IF ] ⋜₀[ IF ] 𝟏[ IF ]
+  γ₂ = 𝟏[ IF ] , (β , γ)
+        where
+         abstract
+          β : 𝟎[ IF ] ∧[ IF ] 𝟏[ IF ] ≡ 𝟎[ IF ]
+          β = 𝟎-left-annihilator-for-∧ IF 𝟏[ IF ]
+
+          γ : 𝟏[ IF ] ∨[ IF ] 𝟏[ IF ] ≡ 𝟏[ IF ]
+          γ = 𝟏-right-annihilator-for-∨ IF 𝟏[ IF ]
+
+  γ₁ : 𝟎[ IF ] ⋜₀[ IF ] 𝟏[ IF ]
+  γ₁ = 𝟎[ IF ] , (β , γ)
+        where
+         abstract
+          β : 𝟎[ IF ] ∧[ IF ] 𝟎[ IF ] ≡ 𝟎[ IF ]
+          β = 𝟎-right-annihilator-for-∧ IF 𝟎[ IF ]
+
+          γ : 𝟏[ IF ] ∨[ IF ] 𝟎[ IF ] ≡ 𝟏[ IF ]
+          γ = 𝟏-left-annihilator-for-∨ IF 𝟎[ IF ]
+
+  𝟎-is-not-𝟏 : ¬ (𝟎[ IF ] ≡ 𝟏[ IF ])
+  𝟎-is-not-𝟏 p = γ
+   where
+    γ : ⊥Ω holds
+    γ = transport _holds (𝟏[ IF ] ≡⟨ p ⁻¹ ⟩ 𝟎[ IF ] ≡⟨ 𝟎-of-IF-is-⊥ ua ⟩ ⊥Ω ∎) *
+
+  ε : ¬ ((U V : ⟨ IF ⟩) → is-prop (well-inside₀ IF U V))
+  ε ψ = 𝟎-is-not-𝟏 (pr₁ (from-Σ-≡ δ))
+   where
+    δ : γ₁ ≡ γ₂
+    δ = ψ 𝟎[ IF ] 𝟏[ IF ] γ₁ γ₂
+
+-- Because well inside is not propositional, we have to truncate it to get a
```

Comment:
This comment should be implemented by begin and end code.

Reply:
Done ✅